### PR TITLE
Auto publish to NPM?

### PR DESCRIPTION
**Pull request by hx-markterry**
#### What does this PR do? (please provide any background)

For our public NPM modules, it would be nice to auto publish to NPM and create a github release for the new version.
#### How can this be tested?

Via CI.
#### Any tech debt?

Although this is mostly a duplication of the private release script, I don't see the need to share code until we have a 3rd copy.
#### What gif best describes how you feel about this work?

giphy
- [ ] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

---

**Reviewer**
- [ ] :+1:
- [ ] I don't think this PR needs any additional reviewers

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

---

### CODE REVIEW

File: nodeApps/createNpmRelease.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+fi
+echo "Message: ${MESSAGE}"
+
+CURRENT_SHA=`git rev-parse HEAD`
+echo "Latest commit is: ${CURRENT_SHA}"
+
+# Tag this deploy and make a release in github
+RELEASE_JSON="{\"tag_name\": \"v${THIS_VERSION}\",\"target_commitish\": \"${CURRENT_SHA}\",\"name\": \"v${THIS_VERSION}\",\"body\": \"${MESSAGE}\",\"draft\": false,\"prerelease\": false}"
+echo "Release JSON: ${RELEASE_JSON}"
+#curl -u "${GITHUB_USER}:${GITHUB_API_TOKEN}" --data "${RELEASE_JSON}" https://api.github.com/repos/${APP_NAME}/releases
```
Comment (notABluesSinger) [inline nodeApps/createNpmRelease.sh]: Any reason you're not currently performing the github release?

File: nodeApps/createNpmRelease.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+LAST_TAG=`git tag --list | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort | tail -n 1`
+echo "Last tag: ${LAST_TAG}"
+THIS_VERSION=`cat package.json | grep version | head -n 1 | cut -d '"' -f 4`
+echo "This version: ${THIS_VERSION}"
+if [ "v${THIS_VERSION}" == "${LAST_TAG}" ]; then
+  echo "WARNING: Version not updated"
+  exit 0
+fi
+
+git checkout ${RELEASE_BRANCH}
```
Comment (notABluesSinger) [inline nodeApps/createNpmRelease.sh]: If this is run via ci and will exit if not the release branch, why do we need to checkout here?

**Comment by pmcnr-hx** [comment]
Is there a compelling reason for writing writing this script in bash rather than node javascript? If we're calling `npm` from it, we'll clearly have a node interpreter available.
Having a more expressive scripting language would make the script simpler and we have more JS expertise than bash in house.
In this particular case I feel we're sticking to bash just for historical reasons.

**Comment by pmcnr-hx** [comment]
It seems the `NPM_TOKEN` environment variable should do the trick and avoid the need for the other credentials (I assume it also works for public modules): http://blog.npmjs.org/post/118393368555/deploying-with-npm-private-modules.

File: nodeApps/createNpmRelease.sh (hunk 1 above)
Comment (hx-markterry) [inline nodeApps/createNpmRelease.sh]: just whilst i'm debugging this as i know this part works from the other script

File: nodeApps/createNpmRelease.sh (hunk 2 above)
Comment (hx-markterry) [inline nodeApps/createNpmRelease.sh]: we probably do not need to do this any more as we are not commiting to the branch in this script, i can remove it.

**Comment by hxpaul** [comment]
if you revisit this and want to rewrite in js you can probably lift most of it from here https://github.com/holidayextras/deployment-helpers/blob/master/nodeApps/generateSignedFile.js

**Pull request closed without merging**